lib/agent0/agent0/hyperdrive/policies/lpandarb.py

```diff
@@ -0,0 +1,642 @@
+"""Agent policy for LP trading that also arbitrage on the fixed rate."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from fixedpointmath import FixedPoint
+
+from agent0.hyperdrive.state import HyperdriveActionType, HyperdriveMarketAction
+from elfpy.types import MarketType, Trade
+
+from .hyperdrive_policy import HyperdrivePolicy
+
+if TYPE_CHECKING:
+    from ethpy.hyperdrive import HyperdriveInterface
+    from numpy.random._generator import Generator as NumpyGenerator
+
+    from agent0.hyperdrive.state import HyperdriveWallet
+
+import time
+
+# pylint: disable=too-many-parameters, too-many-local-variables
+
+# constants
+fp0 = FixedPoint(0)
+fp1 = FixedPoint(1)
+fp2 = FixedPoint(2)
+fp12 = FixedPoint(12)
+fp_seconds_in_year = FixedPoint(365 * 24 * 60 * 60)
+tolerance = 1e-18
+MAX_ITER = 10
+MIN_TRADE_AMOUNT = FixedPoint(scaled_value=20)
+
+
+# functions
+def calc_bond_reserves(
+    share_reserves: FixedPoint,
+    share_price: FixedPoint,
+    target_rate: FixedPoint,
+    position_duration: FixedPoint,
+    time_stretch: FixedPoint,
+):
+    """Calculate the amount of bonds that hit the target target rate for the given shares.
+
+    Parameters
+    ----------
+    share_reserves : FixedPoint
+        The amount of share reserves.
+    share_price : FixedPoint
+        The price of the share.
+    target_rate : FixedPoint
```

Comment:
a lot of the codegen comments are not very helpful, e.g.:
```
target_rate : FixedPoint
    The target rate.
```
Can you give all of them a once-over and update them to have meaningful descriptions? A lot of these are also arguments elsewhere, so you can easily find descriptions in other parts of the code. e.g. in the apy:
```
target_rate : FixedPoint
    The target apr for which to calculate the bond reserves given the pools current share
    reserves.
```